torchgen/gen_functionalization_type.py

```diff
@@ -618,7 +631,9 @@ def emit_inplace_functionalization_body(
       }}
       {unwrap_tensor_args_str}
       if (!({check_all_mutated_args_are_functional})) {{
-        if (({check_any_non_mutated_args_are_functional})) {{
+        // We want to disable this check if there are any XLA tensors.
+        // cpu_tensor.copy_(xla_tensor) is valid code.
+        if ((!{check_any_non_mutated_tensors_are_xla}) && ({check_any_non_mutated_args_are_functional})) {{
```

Comment:
It looks like you forget to add `()` to wrap the expression after `!`. This line should actually be:
```
if ((!({check_any_non_mutated_tensors_are_xla})) && ({check_any_non_mutated_args_are_functional})) {{
```

After the above change, I verified that it fixes our xla test failures:
```
PJRT_DEVICE=CPU python test/test_torch.py -v TestDevicePrecisionXLA.test_copy_noncontig_xla
```

Thanks, Brian!